server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -303,6 +303,49 @@ public String toString() {
         }
     }
 
+    /**
+     * Load {@code float}s from {@code _source}.
+     */
+    public static class DenseVectorBlockLoader extends SourceBlockLoader {
+        private final int dimensions;
+
+        public DenseVectorBlockLoader(ValueFetcher fetcher, LeafIteratorLookup lookup, int dimensions) {
+            super(fetcher, lookup);
+            this.dimensions = dimensions;
+        }
+
+        @Override
+        public Builder builder(BlockFactory factory, int expectedCount) {
+            return factory.denseVectors(expectedCount, dimensions);
+        }
+
+        @Override
+        public RowStrideReader rowStrideReader(LeafReaderContext context, DocIdSetIterator iter) {
+            return new DenseVectors(fetcher, iter);
+        }
+
+        @Override
+        protected String name() {
+            return "Floats";
```

Comment:
Again - just checking if this the value we want to return given that the class is called `DenseVectorBlockLoader`?

Reply:
Good catch, I renamed this multiple times but didn't follow on this method. Thanks! afe79f774a12cf0cd5c26fea871acb5736817f14